### PR TITLE
LOK-2148: Add availability percentage query

### PR DESCRIPTION
**Pull request by cgorantla**
## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2148

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Comment by cgorantla** [comment]
will have to consider first observation time for a metric to calculate reachability more accurately if node was added recently, this is currently being worked on https://opennms.atlassian.net/browse/LOK-2188

File: rest-server/src/main/java/org/opennms/horizon/server/service/metrics/QueryService.java [hunk 1]
```diff
@@ -140,6 +145,16 @@ public String getQueryString(Optional<NodeDTO> node, String metricName, Map<Stri
                     }
             }
         }
+
+        if (isCustomQuery(metricName)) {
```
Comment (freddy61025) [inline rest-server/src/main/java/org/opennms/horizon/server/service/metrics/QueryService.java:149]: nit: the name of the function is not self descriptive. Also may be connect with the last if condition with else if will make more sense.

File: rest-server/src/main/java/org/opennms/horizon/server/service/metrics/QueryService.java [hunk 2]
```diff
@@ -196,6 +211,11 @@ public static Optional<Duration> getDuration(Integer timeRange, TimeRangeUnit ti
         return Optional.empty();
     }
 
+    public static long numofMinutesInDuration(Integer timeRange, TimeRangeUnit timeRangeUnit) {
```
Comment (bouff) [inline rest-server/src/main/java/org/opennms/horizon/server/service/metrics/QueryService.java:214]: Nit: `numOfMinutes...`

Comment (freddy61025) [inline rest-server/src/main/java/org/opennms/horizon/server/service/metrics/QueryService.java:214]: should it be "Of" ?

**Review by freddy61025 — APPROVED** [review]
some minor comments but still fine to merge

**Review by bouff — APPROVED** [review]

**Pull request merged**